cloudant-sync-datastore-core/src/test/java/com/cloudant/http/HttpTest.java

```diff
@@ -40,6 +43,36 @@
 public class HttpTest extends CouchTestBase {
 
     private String data = "{\"hello\":\"world\"}";
+    private ByteArrayInputStream bis;
+
+    @Before
+    public void setupDataByteStream() {
+        this.bis = new ByteArrayInputStream(data.getBytes());
+    }
+
+    private HttpConnection postAndAssertNothingReadBeforeSettingBodyGenerator(CouchConfig config)
+            throws Exception {
+        HttpConnection conn = new HttpConnection("POST", config.getRootUri().toURL(),
+                "application/json");
+
+        // nothing read from stream
+        Assert.assertEquals(bis.available(), data.getBytes().length);
+
+        conn.setRequestBody(new HttpConnection.InputStreamGenerator() {
+            @Override
+            public InputStream getInputStream() {
+                return bis;
+            }
+        });
+        // This test invokes HttpConnection directly rather than via the CouchClient, so we need to
+        // force the addition of some default interceptors, which we obtain reflectively for the
+        // purposes of these tests.
+        Field interceptors = CouchClient.class.getDeclaredField("DEFAULT_REQUEST_INTERCEPTORS");
```

Comment:
Since `CouchClient` is internal, I don't mind the field `DEFAULT_REQUEST_INTERCEPTORS` being public if we are going to use it in a few other places, even if that's in test code - but it's up to you.

Reply:
Findbugs flagged it as malicious because it was `public` and a mutable list, but now that I've changed the initialization the list is no longer mutable, so happy to make it public again.
Done in ae17649d66c5827add7df0d5d7b75c2d13a5d11d